[conex] Use a custom cc_library rule for vendoring

Reasoning for not upstreaming this patch: Drake-specific vendoring.

--- conex/BUILD
+++ conex/BUILD
@@ -1,3 +1,8 @@
+load(
+    "@drake//tools/workspace/conex_internal:vendor.bzl",
+    cc_library = "conex_cc_library",
+)
+
 cc_library(
     name = "test_util",
     srcs = glob(["test/test_util.cc"]),
